lib/trie/db/db.go

```diff
@@ -30,16 +53,46 @@ func NewMemoryDBFromProof(encodedNodes [][]byte) (*MemoryDB, error) {
 
 }
 
-func (mdb *MemoryDB) Get(key []byte) (value []byte, err error) {
-	if len(key) < common.HashLength {
-		return nil, fmt.Errorf("expected %d bytes length key, given %d (%x)", common.HashLength, len(key), value)
+func (mdb *MemoryDB) Copy() Database {
+	newDB := NewEmptyMemoryDB()
+	copyData := make(map[common.Hash][]byte, len(mdb.data))
+
+	for k, v := range mdb.data {
+		copyData[k] = v
+	}
+
+	newDB.data = copyData
+	return newDB
+}
+
+func (mdb *MemoryDB) Get(key []byte) ([]byte, error) {
+	if len(key) != common.HashLength {
+		return nil, fmt.Errorf("expected %d bytes length key, given %d (%x)", common.HashLength, len(key), key)
 	}
 	var hash common.Hash
 	copy(hash[:], key)
```

Comment:
Is this copy needed? I would suggest using the key directly. We will not change the `key`, we just use it to get a value from a map

```
if value, found := mdb.data[key]; found {
	return value, nil
}
```

Reply:
The copy is to convert the `key` into a `common.Hash` since data is `data  map[common.Hash][]byte`

I think and alternative is to create the Hash using the key, like: `hashedKey := common.Hash(key)`